### PR TITLE
Validate the tx_json field in sign_for (RIPD-1100)

### PR DESCRIPTION
**Pull request by nbougalis**
(targeted against *release* but should be merged into both *release* and *develop*)

### CODE REVIEW

File: src/ripple/rpc/tests/JSONRPC.test.cpp [hunk 1]
```diff
@@ -1112,6 +1112,45 @@ R"({
 "Missing field 'tx_json.TransactionType'.",
 "Missing field 'tx_json.TransactionType'."}},
 
+{ "Invalid field 'tx_json': not an object",
```
Comment (miguelportilla) [inline src/ripple/rpc/tests/JSONRPC.test.cpp]: Suggestion: Change comment to "Invalid field 'tx_json': string" or change tx_json value to "null".

**Comment by miguelportilla** [comment]
Left a comment, not holding back. :+1:

**Comment by scottschurr** [comment]
Looks great.  I looked for a more elegant solution and didn't find one.  Nice tests.  :+1:.

**Comment by nbougalis** [comment]
Merged into *develop* as 8dbad6215304b3dff5a0545b2363405b59ab5bcd and 60ff83f280eadae4f7a9ade4aaf3ffc87df5a561. Merge to *release* pending.

**Pull request closed without merging**